libs/community/langchain_community/document_loaders/upstage.py

```diff
@@ -0,0 +1,227 @@
+    ELEMENT = "element"
+    PAGE = "page"
+
+
+def validate_api_key(api_key: str) -> None:
+    """
+    Validates the provided API key.
+
+    Args:
+        api_key (str): The API key to be validated.
+
+    Raises:
+        ValueError: If the API key is empty or None.
+
+    Returns:
+        None
+    """
+    if not api_key:
+        raise ValueError("API Key is required for Upstage Document Loader")
+
+
+def validate_file_path(file_path: str) -> None:
+    """
+    Validates if a file exists at the given file path.
+
+    Args:
+        file_path (str): The path to the file.
+
+    Raises:
+        FileNotFoundError: If the file does not exist at the given file path.
+    """
+    if not os.path.exists(file_path):
+        raise FileNotFoundError(f"File not found: {file_path}")
+
+
+def parse_output(data: dict, output_type: str) -> str:
+    """
+    Parse the output data based on the specified output type.
+
+    Args:
+        data (dict): The data to be parsed.
+        output_type (str): The type of output to parse.
+
+    Returns:
+        str: The parsed output.
+
+    Raises:
+        ValueError: If the output type is invalid.
+    """
+    if (output_type) == OutputType.TEXT.value:
+        return data["text"]
+    elif (output_type) == OutputType.HTML.value:
+        return data["html"]
+    else:
+        raise ValueError(f"Invalid output type: {output_type}")
+
+
+class UpstageDocumentLoader(BaseLoader):
+    """
+    A document loader for Upstage API.
+
+    Args:
+        file_path (str): The path to the file to be loaded.
+        output_type (str, optional): The desired output type.
+                                     Defaults to OutputType.TEXT.value.
+        split (str, optional): The split type for the document.
+                               Defaults to SplitType.NONE.value.
+        api_key (str, optional): The API key for authentication. Defaults to "".
+
+    Attributes:
+        file_path (str): The path to the file to be loaded.
+        output_type (str): The desired output type.
+        split (str): The split type for the document.
+        api_key (str): The API key for authentication.
+        file_name (str): The name of the file.
+
+    Raises:
+        ValueError: If the API call returns a non-200 status code.
+        ValueError: If an invalid split type is provided.
+
+    """
+
+    def __init__(
+        self,
+        file_path: str,
+        output_type: str = OutputType.TEXT.value,
+        split: str = SplitType.NONE.value,
+        api_key: str = "",
+        url: str = LAYOUT_ANALYZER_URL,
+    ):
+        """Initialize with the file path."""
+        self.file_path = file_path
+        self.output_type = output_type
+        self.split = split
+        self.api_key = api_key
+        self.file_name = os.path.basename(file_path)
+        self.url = url
+
+        validate_file_path(self.file_path)
+        validate_api_key(self.api_key)
+
+    def load(self) -> List[Document]:
```

Comment:
Define lazy_load rather than load

Reply:
I apologize, but I'm afraid I'm also unclear about the meaning of your comment. Could you please provide more detail? Specifically, are you asking if I simply need to rename the `load` function to `lazy_load` or if there are additional steps required to define the `lazy_load` function?

Reply:
Just as you suggested, I added the lazy\_load function.